File: ai-server/app/models/product_snapshot.py
from sqlalchemy import Column, BigInteger, ForeignKey
from sqlalchemy import Integer, Float, Text, TIMESTAMP
from sqlalchemy.sql import func

from app.models.base import Base


class ProductSnapshot(Base):

    __tablename__ = "product_snapshot"

    id = Column(BigInteger, primary_key=True)

    product_id = Column(
        BigInteger,
        ForeignKey("product.id")
    )

    price = Column(Integer)

    rating = Column(Float)

    review_count = Column(Integer)

    seller_name = Column(Text)

    crawled_at = Column(TIMESTAMP, server_default=func.now())

File: api-server/src/main/java/com/productresearch/apiserver/domain/product/entity/ProductSnapshot.java
package com.productresearch.apiserver.domain.product.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "product_snapshot")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@AllArgsConstructor
public class ProductSnapshot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Product product;

    private Integer price;
    private Double rating;

    @Column(name = "review_count")
    private Integer reviewCount;

    @Column(name = "seller_name")
    private String sellerName;

    @Column(name = "crawled_at")
    private LocalDateTime crawledAt;

    @PrePersist
    protected void onCreate() {
        crawledAt = LocalDateTime.now();
    }
}